lldb/source/Plugins/Process/gdb-remote/ProcessGDBRemote.h

```diff
@@ -440,6 +440,12 @@ class ProcessGDBRemote : public Process,
   void HandleStopReply() override;
   void HandleAsyncStructuredDataPacket(llvm::StringRef data) override;
 
+  // Handle thread specific async interrupt and return the original thread
+  // that requested the async interrupt. It can be null if original thread
+  // has exited.
+  lldb::ThreadSP HandleThreadAsyncInterrupt(uint8_t signo,
```

Comment:
I can't see any reason why the ability to manage this interrupt event is GDBRemote specific.  So this should be a virtual method in Process, and overridden here.